File: leetcode/0283-move-zeroes/solution.java
class Solution {
    public void moveZeroes(int[] nums) {
        int i=0;
        int[] ans = new int[nums.length];
        for(int n: nums){
            if(n!=0){
                ans[i++] = n;
            } 
        }

        for(int j=0;j<nums.length;j++){
            if(j<i){
                nums[j]=ans[j];
            } else{
                nums[j]=0;
            }
        }
    }
}

File: leetcode/0283-move-zeroes/solution.py
class Solution(object):
    def moveZeroes(self, nums):
        ph , se = 0 ,0

        while se < len(nums):
            if nums[se] != 0:
                nums[se], nums[ph] = nums[ph],nums[se]
                ph += 1
                
            se += 1
            
        
